### PR TITLE
Remove admin write permissions on system vault

### LINKED ISSUE #8269: Kill `write` permission for admin

### PR DESCRIPTION
**Pull request by flvndvd**
## Description

This PR fixes https://github.com/dust-tt/dust/issues/8269.

This PR removes the "write" permission for admin on the system vault and refactors some permission logic to consolidate around `canAdministrate`. This need is now covered by the "admin" permission.

### CODE REVIEW

File: front/pages/api/w/[wId]/vaults/[vId]/data_sources/[dsId]/index.ts [hunk 1]
```diff
@@ -49,7 +49,7 @@ async function handler(
     });
   }
 
-  if (vault.isSystem() && !auth.isAdmin()) {
+  if (vault.isSystem() && !vault.canAdministrate(auth)) {
```
Comment (Fraggle) [inline front/pages/api/w/[wId]/vaults/[vId]/data_sources/[dsId]/index.ts:52]: do you need both checks ?

File: front/pages/api/w/[wId]/vaults/[vId]/data_sources/[dsId]/index.ts [hunk 2]
```diff
@@ -127,8 +116,9 @@ async function handler(
     case "DELETE": {
       const isAuthorized =
         vault.canWrite(auth) ||
+        // Only allow to delete Snowflake connectors if the user is an admin.
         (vault.isSystem() &&
-          auth.isAdmin() &&
+          vault.canAdministrate(auth) &&
```
Comment (Fraggle) [inline front/pages/api/w/[wId]/vaults/[vId]/data_sources/[dsId]/index.ts:121]: same.

File: front/pages/api/w/[wId]/vaults/[vId]/data_sources/index.ts [hunk 3]
```diff
@@ -113,7 +114,7 @@ async function handler(
   }
 
   if (space.isSystem()) {
-    if (!auth.isAdmin()) {
+    if (!space.canAdministrate(auth)) {
```
Comment (Fraggle) [inline front/pages/api/w/[wId]/vaults/[vId]/data_sources/index.ts:117]: same.

**Review by Fraggle — APPROVED** [review]

File: front/pages/api/w/[wId]/vaults/[vId]/data_sources/[dsId]/index.ts (hunk 1 above)
Comment (flvndvd) [reply front/pages/api/w/[wId]/vaults/[vId]/data_sources/[dsId]/index.ts:52]: The rule is that for the system vault no one can write, so we use the verb `administrate`. Where `administrate` means:
- Connecting managed data sources
- Administrating vaults (name, members, ...)

**Pull request merged**